controllers/mentor.go

```diff
@@ -7,40 +7,115 @@ import (
 	"net/http"
 	"os"
 	"strings"
+	"fmt"
 
 	"kwoc20-backend/models"
+	logs "kwoc20-backend/utils/logs/pkg"
 
 	"github.com/jinzhu/gorm"
+	"github.com/go-kit/kit/log/level"
 )
 
-// MentorOauth Handler for Github OAuth of Mentor
+// MentorOAuth Handler for Github OAuth of Mentor
 func MentorOAuth(w http.ResponseWriter, r *http.Request) {
 	// get the code from frontend
 	var mentorOAuth1 interface{}
-	body, _ := ioutil.ReadAll(r.Body)
-	err := json.Unmarshal(body, &mentorOAuth1)
-	mentorOAuth, _ := mentorOAuth1.(map[string]interface{})
+	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte(`{"message": "` + err.Error() + `"}`))
+		_, writeErr := w.Write([]byte(`{"message": "` + err.Error() + `"}`))
+		if writeErr != nil {
+			logErr := level.Warn(logs.Logger).Log("error", fmt.Sprintf("%v",writeErr))
+			if logErr != nil {
+				panic("Log Error")
+			}
+		}
+		logErr := level.Error(logs.Logger).Log("error", fmt.Sprintf("%v",err))
+		if logErr != nil {
```

Comment:
Also try avoiding(not use) using the panics, generally if our website goes down due to a simple issue of Logging not working, it would be unnecessarily useless to panic, where we could just log and move forward 🙃

More info https://stackoverflow.com/questions/44504354/should-i-use-panic-or-return-error